### DISCUSSION TITLE
How to disable migrations?

### DISCUSSION THREAD

**Discussion by VictorioBerra** [pocketbase/pocketbase#4400] [Q&A]
I am using wgo for auto-reload: https://github.com/bokwoon95/wgo

`go install github.com/bokwoon95/wgo@latest`

`wgo run -file .html ./main.go serv`

Id like to speed up my reloads. Is there a `--no-migrate` for Pocketbase?

**Comment by ganigeorgiev** [comment]
It is not possible at the moment to disable the automatic unappplied migrations execution on serve.

> Id like to speed up my reloads

I'm not sure that I understand what do you mean by this. If you mean the build performance then I don't think migrations have any significant effect on the build or startup times.
PocketBase builds could be slow most likely due to the linking of the SQLite library (CGO or pure Go) and other larger dependencies (ex. `aws-sdk-go`). Build performance could be also affected by the running OS (if you use Windows you can try to add an exception for the wgo and Go toolchain in your antrivirus software)

**Reply by VictorioBerra** [reply]
Funny you mention the antivirus because it has already leaped up and blocked a few go related things. I have since excluded all the go folders.

> > Id like to speed up my reloads

> I'm not sure that I understand what do you mean by this.

I am new with Go, I come from ASPNET Core and NodeJS platforms where I can edit .js files and .html templates and I have near instant "reloads" where I can see my new content and test my changes as I develop. This is, in my IMO, one of the most important things for developing web apps.

Does go have hot reload? AOT? browser sync? Anything to speed up changes and allow me to develop faster?

**Reply by erikd234** [reply]
Go doesnt have not reload, But if you want a light wait, watch file changes tool I made one exactly for my pocketbase project. You can run tests or the go run main.go serve to watch when files change and then reload the go code.

There are other alternatives out there too that have more features than my tool.

`ifchange -cmd "go run main.go serve"` -only ".go"`

https://github.com/erikd234/ifchange

**Reply by IgnisDa** [reply]
> It is not possible at the moment to disable the automatic unappplied migrations execution on serve.
> 
> > Id like to speed up my reloads
> 
> I'm not sure that I understand what do you mean by this. If you mean the build performance then I don't think migrations have any significant effect on the build or startup times. PocketBase builds could be slow most likely due to the linking of the SQLite library (CGO or pure Go) and other larger dependencies (ex. `aws-sdk-go`). Build performance could be also affected by the running OS (if you use Windows you can try to add an exception for the wgo and Go toolchain in your antrivirus software)

@ganigeorgiev Is this possible now? If not, would you be interested in a PR for this?

**Reply by ganigeorgiev** [reply]
@IgnisDa Not out of the box because I don't see the point of it. Feel free to elaborate what is the problem with the current setup and what you are trying to do but I'm not willing to accept PRs if I don't understand the use case for it.

**Reply by IgnisDa** [reply]
My usecase is pretty simple. I do not want the gear icon here to be available at all:

<img width="347" alt="image" src="https://github.com/user-attachments/assets/f74c4235-a28d-45dd-8780-e3bceef94b0f" />

Reason: I always create migrations locally and then push them to prod. I do not want that someone mistakenly makes a schema change on prod since then the local environments will have no way of tracking that schema change. For example, if i remove a column in a collection on prod, but the local code wont be able to account for this change.

**Reply by ganigeorgiev** [reply]
I don't see what this has to do with the migrations (or this discussion in general).

You can hide the icon from the _Dashboard > Settings > "Hide collection create and edit controls"_.
